src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Receiver.java

```diff
@@ -0,0 +1,372 @@
+            dest.getBuffer()[4] ^= source.getBuffer()[4];
+            dest.getBuffer()[5] ^= source.getBuffer()[5];
+            dest.getBuffer()[6] ^= source.getBuffer()[6];
+            dest.getBuffer()[7] ^= source.getBuffer()[7];
+
+            // Skip the 9th to 12th bytes of the header.
+        }
+
+        /**
+         * Xor the payloads of the source and destination
+         * @param source the packet to xor the payload from
+         * @param sourceOffset the offset in the source packet at which the
+         * payload begins.  Note that this is not necessarily the location
+         * at which the RTP payload starts...for the purpose of FlexFEC,
+         * everything after the fixed RTP header is considered the 'payload'
+         * @param dest the packet to xor the payload into
+         * @param destOffset the offset in the dest packet at which the payload
+         * begins
+         * @param payloadLength the length of the source's payload
+         */
+        private void xorPayloads(RawPacket source, int sourceOffset,
+                                 RawPacket dest, int destOffset,
+                                 int payloadLength)
+        {
+            for (int i = 0; i < payloadLength; ++i)
+            {
+                dest.getBuffer()[destOffset + i] ^=
+                    source.getBuffer()[sourceOffset + i];
+            }
+        }
+
+        /**
+         * Do the final work when recovering an RTP packet (set the RTP version,
+         * the length, the sequence number, and the ssrc)
+         * @param fecPacket the fec packet
+         * @param recoveredPacket the media packet which was recovered
+         */
+        private boolean finishPacketRecovery(FlexFec03Packet fecPacket, RawPacket recoveredPacket)
+        {
+            // Set the RTP version to 2.
+            recoveredPacket.getBuffer()[0] |= 0x80; // Set the 1st bit
+            recoveredPacket.getBuffer()[0] &= 0xbf; // Clear the second bit
+
+            // Recover the packet length, from temporary location.
+            int length = RTPUtils.readUint16AsInt(recoveredPacket.getBuffer(), 2);
+            if (length > recoveredPacket.getBuffer().length)
+            {
+                logger.error("Length field of recovered packet is larger" +
+                    " than its buffer");
+                return false;
+            }
+            // The length field used in the xor does not include the header
+            // length, but we want to include the fixed header length when
+            // setting the length on the packet object
+            recoveredPacket.setLength(length + RawPacket.FIXED_HEADER_SIZE);
+
+            // Set the SN field.
+            RTPUtils.writeShort(recoveredPacket.getBuffer(), 2,
+                (short)missingSequenceNumber);
+
+            // Set the SSRC field.
+            RTPUtils.writeInt(recoveredPacket.getBuffer(), 8,
+                (int)fecPacket.getProtectedSsrc());
```

Comment:
We have setSequenceNumber and setSSRC helpers

Reply:
cool, fixed